lib/Chat/Parser/UserMention.php

```diff
@@ -90,8 +90,14 @@ protected function parseMessage(Message $chatMessage): void {
 			return mb_strlen($m2['id']) <=> mb_strlen($m1['id']);
 		});
 
+		$actor = $this->participantService->getParticipantByActor($chatMessage->getRoom(), $chatMessage->getActorType(), $chatMessage->getActorId());
+		$onlyModsCanMentionAll = $chatMessage->getRoom()->getMentionPermissions() === Room::MENTION_PERMISSIONS_MODERATORS;
 		foreach ($mentions as $mention) {
 			if ($mention['type'] === 'user' && $mention['id'] === 'all') {
+				if ($onlyModsCanMentionAll && !$actor->hasModeratorPermissions()) {
```

Comment:
Not sure about this. I think the most important bit is not doing notifications and marking the room as mentioned in the list. but it can also be quite confusing if you see the `@all` rendered and you dont know others were not notified.
But it also retroactively changes previous messages.

But I guess it's fine for now.

Reply:
> But it also retroactively changes previous messages.

I was worried about that too, but was not able to find any straight forward way to prevent it

Reply:
We have a meta-data field since 29:
https://github.com/nextcloud/spreed/blob/1ccbf67da231c37801c1406b10000144125aec35/lib/Chat/ChatManager.php#L323-L327

We could store in this field during posting whether the mention should be rendered.

Reply:
oh great! I will quickly implement it